pkg/patch/patch.go

```diff
@@ -213,28 +353,30 @@ func patchWithContext(ctx context.Context, ch chan error, image, reportFile, rep
 	}
 	solveOpt.SourcePolicy, err = build.ReadSourcePolicy()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	if solveOpt.SourcePolicy != nil {
 		switch {
 		case strings.Contains(solveOpt.SourcePolicy.Rules[0].Updates.Identifier, "redhat"):
 			err = errors.New("RedHat is not supported via source policies due to BusyBox not being in the RHEL repos\n" +
 				"Please use a different RPM-based image")
-			return err
+			return nil, err
 
 		case strings.Contains(solveOpt.SourcePolicy.Rules[0].Updates.Identifier, "rockylinux"):
 			err = errors.New("RockyLinux is not supported via source policies due to BusyBox not being in the RockyLinux repos\n" +
 				"Please use a different RPM-based image")
-			return err
+			return nil, err
 
 		case strings.Contains(solveOpt.SourcePolicy.Rules[0].Updates.Identifier, "alma"):
 			err = errors.New("AlmaLinux is not supported via source policies due to BusyBox not being in the AlmaLinux repos\n" +
 				"Please use a different RPM-based image")
-			return err
+			return nil, err
 		}
 	}
 
+	// Create a channel to receive the patched image digest
+	patchedImageDigest := make(chan string, 1)
```

Comment:
why do we need a channel here?

Reply:
This channel ferries the digest from the go routine out into the larger context to avoid races, but I've just realised since we use `eg.Wait`, there won't be any race, so I've changed it to be just a regular string var